File: Leetcode/78.cpp
/**
 * 78. 子集 Cascading
 * link：https://leetcode-cn.com/problems/subsets/
 */

class Solution
{
public:
    vector<vector<int>> subsets(vector<int> &nums)
    {
        vector<vector<int>> result;
        result.push_back(vector<int>());

        for (const auto &num : nums)
        {
            int size = result.size();
            for (int i = 0; i < size; i++)
            {
                vector<int> cur(result[i]);
                cur.push_back(num);
                result.push_back(cur);
            }
        }
        return result;
    }
};

File: Leetcode/78.py
# 78. 子集 Cascading
# link：https://leetcode-cn.com/problems/subsets/

class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        result: List[List[int]] = []
        result.append([])

        for num in nums:
            for i in range(0, len(result)):
                result.append(result[i]+[num])

        return result
